src/screens/ChannelListScreen/elements/Search.js

```diff
@@ -94,18 +96,19 @@ const styles = StyleSheet.create({
     alignSelf: 'center',
     justifyContent: 'center'
   },
-  newPostText: {
-    marginRight: dimen.normalizeDimen(5),
+  newPostText: (dynamicColors) => ({
+    color: dynamicColors.primary,
+    marginRight: 11,
     fontFamily: 'Inter-SemiBold',
     fontSize: 12,
     lineHeight: 14.52
-  },
+  }),
   newChatIcon: {
     marginTop: 0
   },
   animatedViewContainer: {
     flexDirection: 'row',
-    backgroundColor: 'white',
+    backgroundColor: COLORS.white,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `newPostText` is defined within the `styles` object. This is not a good practice as it mixes styling with logic. Consider moving this function outside of the `styles` object and call it when needed.

```diff
- newPostText: (dynamicColors) => ({
-   color: dynamicColors.primary,
-   marginRight: 11,
-   fontFamily: 'Inter-SemiBold',
-   fontSize: 12,
-   lineHeight: 14.52
- }),

+ const getNewPostTextStyle = (dynamicColors) => ({
+   color: dynamicColors.primary,
+   marginRight: 11,
+   fontFamily: 'Inter-SemiBold',
+   fontSize: 12,
+   lineHeight: 14.52
+ });
```

Then use it like this:

```javascript
<Text style={getNewPostTextStyle(dynamicColors)}>
  {StringConstant.chatTabHeaderCreateChatButtonText}
</Text>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->